cmd/mdatagen/internal/samplereceiver/generated_component_telemetry_test.go

```diff
@@ -0,0 +1,72 @@
+// Code generated by mdatagen. DO NOT EDIT.
+
+package samplereceiver
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/receivertest"
+)
+
+type componentTestTelemetry struct {
```

Comment:
would it make sense to provide a "Reset" function here, so that multiple tests can exist without re-creating test settings for each case?

Reply:
Happy to add this as needed. I suspect it will be needed sooner than later but this particular PR doesn't need it just yet